### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by matthiaskrgr**
Fixes

clippy::single_char_pattern
clippy::manual_strip
clippy::unneccessary_wraps
clippy::redundant_closure
clippy::single_char_add_str
clippy::search_is_some
clippy::clone_on_copy
clippy::into_iter_on_ref

### CODE REVIEW

**Comment by calebcartwright** [comment]
Thanks for this. Was there another lint rule in there that maybe got lost in a copy/paste? You had both `unnecessary_wraps` and `unnecessary_wrap` in the issue description, not sure if that's a typo?

**Comment by matthiaskrgr** [comment]
I glanced over the changes again and I didn't find any lints that look like they are not listed.
There `unnecessary_wrap` was renamed to `unnecessary_wraps` recently inside clippy so I probably got confused. :/

**Comment by calebcartwright** [comment]
> I glanced over the changes again and I didn't find any lints that look like they are not listed.
> There `unnecessary_wrap` was renamed to `unnecessary_wraps` recently inside clippy so I probably got confused. :/

No worries. I only took a quick glance through and remember seeing a couple changes that my mental model couldn't map back to the lint list, but will take a closer look in the near future

File: src/formatting/expr.rs [hunk 1]
```diff
@@ -444,7 +444,7 @@ pub(crate) fn format_expr(
     };
 
     expr_rw
-        .and_then(|expr_str| recover_comment_removed(expr_str, expr.span, context))
+        .map(|expr_str| recover_comment_removed(expr_str, expr.span, context))
```
Comment (calebcartwright) [inline src/formatting/expr.rs:447]: this was the one that confused me on my initial cursory glance as I thought both were `Option<String>`'s but see now that the signature on `recover_comment_removed` has been changed :+1:

File: src/formatting/vertical.rs [hunk 2]
```diff
@@ -50,7 +50,7 @@ impl AlignedItem for ast::StructField {
             mk_sp(self.attrs.last().unwrap().span.hi(), self.span.lo())
         };
         let attrs_extendable = self.ident.is_none() && is_attributes_extendable(&attrs_str);
-        rewrite_struct_field_prefix(context, self).and_then(|field_str| {
+        Some(rewrite_struct_field_prefix(context, self)).and_then(|field_str| {
```
Comment (calebcartwright) [inline src/formatting/vertical.rs:53]: This looks a bit odd. Could we tweak a bit, maybe just create a local with the field str or move the call down below where the corresponding arg is being passed?

File: src/formatting/expr.rs (hunk 1 above)
Comment (matthiaskrgr) [reply src/formatting/expr.rs:447]: Yes, sorry, should have left a comment :/

File: src/formatting/vertical.rs (hunk 2 above)
Comment (matthiaskrgr) [reply src/formatting/vertical.rs:53]: Yeah, ideally, clippy would catch `Some(x).and_then()` :)

Comment (calebcartwright) [reply src/formatting/vertical.rs]: That would be helpful indeed!

**Pull request merged**